src/AI/agents/api/routes/agent.py

```
@@ -88,6 +88,19 @@ async def start_agent(
                 log.error(f"Failed to process attachments for session {req.session_id}: {e}")
                 raise HTTPException(status_code=400, detail=f"Invalid attachment payload: {e}") from e
 
+        # MCP readiness check — live ping to verify MCP is actually reachable
+        from agents.services.mcp import get_mcp_client
+        mcp = get_mcp_client()
+        try:
+            await mcp.check_server_status()
+        except Exception as ping_err:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)